packages/xml-views-validation/src/validate-xml-views.ts

```diff
@@ -0,0 +1,55 @@
+import { flatMap } from "lodash";
+import {
+  XMLDocument,
+  XMLAstVisitor,
+  XMLAttribute,
+  XMLElement,
+  accept
+} from "@xml-tools/ast";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import { UI5XMLViewIssue } from "../api";
+
+export interface UI5Validators {
+  element: ((
```

Comment:
Do we want also document-level validators (for example: to verify that there is a root element, or that the root element type is `mvc:View`)?

Reply:
The base visitor does not distinguish between the root element and other elements.
It could be done, however the root node is actually an "XmlDocument" node
which we can easily add support for, this may be needed for "stable id" validation, but I don't know the scenario well enough for now.

Regarding "root **element** validation": I am not sure, an element validation could check if the element is the Root before performing its checks, so such an API may not be needed. 

Your two examples may(???) not be valid validations, e.g: 
- If there is no root element (empty xml), what are we validating?
- The root element could also be a fragment, or perhaps a fragment could start with any root element? 

I would wait for expending the API until we have more specific scenarios.

Reply:
> The base visitor does not distinguish between the root element and other elements.
> It could be done, however the root node is actually an "XmlDocument" node
> which we can easily add support for, 

Yes, I meant for `XMLDocument`. Since it holds the root element the validation could be on it.

> this may be needed for "stable id" validation, but I don't know the scenario well enough for now.

I think stable IDs validation is not related, it's a validation that an `id` attribute exists (and possibly has a unique value in the document - but this is probably better to do in a different validation) except in some specific cases.

> Regarding "root **element** validation": I am not sure, an element validation could check if the element is the Root before performing its checks, so such an API may not be needed.

It's possible, but less efficient and logically these validations belongs on the document root I think.

> Your two examples may(???) not be valid validations, e.g:

Considering I just made them up it's very possible :)

> * If there is no root element (empty xml), what are we validating?

That the document isn't empty, I guess. I would say that normally this belongs in the generic xml validations but if a regular xml can be empty then it's specific (assuming xml views can't be empty, which should be checked of course, before we add a validation like this).

> * The root element could also be a fragment, or perhaps a fragment could start with any root element?

Yes, you would need to check the file name. But we have the document so maybe we can implement it. I think for fragments the root element is `sap.ui.core.FragmentDefinition`.

> I would wait for expending the API until we have more specific scenarios.

Ok

Reply:
> That the document isn't empty

An empty XML document, or one without a root element is actually a syntax error.
- https://github.com/SAP/xml-tools/blob/master/packages/parser/lib/parser.js#L33

> It's possible, but less efficient and logically these validations belongs on the document root I think.

If checking the top level `XMLDocument` is fine, we can easily add it to the visitor. I agree it is more logical.

> Yes, you would need to check the file name. But we have the document so maybe we can implement it. I think for fragments the root element is sap.ui.core.FragmentDefinition.

Interesting, perhaps additional context would be needed in this case, we need to know if its a fragment or a view, but we can easily compose a different set of check functions for each scenario.